### PR TITLE
Connect chat to mongodb

### PR DESCRIPTION
**Pull request by ZijiaZhang**
- [ ] All features are tested
- [ ] Version is bumped
- [ ] Changelog is updated
- [ ] Documentation is updated if needed

### CODE REVIEW

File: test/backend/api/v1/ChatTest.spec.ts [hunk 1]
```diff
@@ -1,25 +1,86 @@
...
+                    expect(res.body).have.key('message');
+                    expect(res.body.message).eql('sender_id or receiver_id not found');
+                }).catch((err) => {
+                    expect.fail('should not throw error' + err);
+                })
+        });
+    });
+
+
+    describe('Get all chats2', () => {
```
Comment (xddenise) [inline test/backend/api/v1/ChatTest.spec.ts:58]: Why "Get all chats2"? Shouldn't it be something like "Post a chat"?

File: src/backend/api/v1/Chats.ts [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+    const sender_id = req.query.sender_id;
+    const receiver_id = req.query.receiver_id;
+    if (!sender_id || !receiver_id){
+        res.status(400).json({message: 'sender_id or receiver_id not found'});
+        return;
+    }
+
+    const allMessages= Chat.find({senderUsername: sender_id, receiverUsername: receiver_id});
+    allMessages.exec();
+    return allMessages.then((chats: any) =>{
```
Comment (xddenise) [inline src/backend/api/v1/Chats.ts:16]: Should the type for chats be `databaseChat[]`?

File: test/backend/api/v1/ChatTest.spec.ts (hunk 1 above)
Comment (ZijiaZhang) [reply test/backend/api/v1/ChatTest.spec.ts:58]: I forgot to change this when I tried to debug the app & mongoose connection between tests. 😢

File: src/backend/Models.ts [hunk 3]
```diff
@@ -69,6 +69,13 @@ const postSchema = new Schema({
 });
 export const Post = model('post', postSchema);
 
+export interface databaseChat extends Document{
```
Comment (xddenise) [inline src/backend/Models.ts:72]: Do you think interfaces for models should be shared with frontend so parsing is easier and more readable?

**Pull request merged**